diff --git a/lib/automake.mk b/lib/automake.mk
index 17b36b4..3152060 100644
--- a/lib/automake.mk
+++ b/lib/automake.mk
@@ -5,7 +5,10 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
 
-lib_LTLIBRARIES += lib/libopenvswitch.la
+noinst_LTLIBRARIES += lib/libopenvswitch.la
+lib_LTLIBRARIES += lib/libnoiro_openvswitch.la
+lib_libnoiro_openvswitch_la_SOURCES =
+lib_libnoiro_openvswitch_la_LIBADD = lib/libopenvswitch.la
 
 lib_libopenvswitch_la_LIBADD = $(SSL_LIBS)
 lib_libopenvswitch_la_LIBADD += $(CAPNG_LDADD)
@@ -15,7 +18,7 @@ if WIN32
 lib_libopenvswitch_la_LIBADD += ${PTHREAD_LIBS}
 endif
 
-lib_libopenvswitch_la_LDFLAGS = \
+lib_libnoiro_openvswitch_la_LDFLAGS = \
         $(OVS_LTINFO) \
         -Wl,--version-script=$(top_builddir)/lib/libopenvswitch.sym \
         $(AM_LDFLAGS)
@@ -366,8 +369,11 @@ nodist_lib_libopenvswitch_la_SOURCES = \
 	lib/vswitch-idl.h
 CLEANFILES += $(nodist_lib_libopenvswitch_la_SOURCES)
 
-lib_LTLIBRARIES += lib/libsflow.la
-lib_libsflow_la_LDFLAGS = \
+noinst_LTLIBRARIES += lib/libsflow.la
+lib_LTLIBRARIES += lib/libnoiro_sflow.la
+lib_libnoiro_sflow_la_SOURCES =
+lib_libnoiro_sflow_la_LIBADD = lib/libsflow.la
+lib_libnoiro_sflow_la_LDFLAGS = \
         $(OVS_LTINFO) \
         -Wl,--version-script=$(top_builddir)/lib/libsflow.sym \
         $(AM_LDFLAGS)
diff --git a/lib/libopenvswitch.pc.in b/lib/libopenvswitch.pc.in
index 2a3f2ca..5a0eca4 100644
--- a/lib/libopenvswitch.pc.in
+++ b/lib/libopenvswitch.pc.in
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libopenvswitch
 Description: Open vSwitch library
 Version: @VERSION@
-Libs: -L${libdir} -lopenvswitch
+Libs: -L${libdir} -lnoiro_openvswitch
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/lib/libsflow.pc.in b/lib/libsflow.pc.in
index e70a2b7..6f73164 100644
--- a/lib/libsflow.pc.in
+++ b/lib/libsflow.pc.in
@@ -3,9 +3,9 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
-Name: libofproto
+Name: libsflow
 Description: sFlow library of Open vSwitch
 Version: @VERSION@
-Libs: -L${libdir} -lsflow
+Libs: -L${libdir} -lnoiro_sflow
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/ofproto/automake.mk b/ofproto/automake.mk
index 7c08b56..5ae1730 100644
--- a/ofproto/automake.mk
+++ b/ofproto/automake.mk
@@ -5,8 +5,12 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
 
-lib_LTLIBRARIES += ofproto/libofproto.la
-ofproto_libofproto_la_LDFLAGS = \
+noinst_LTLIBRARIES += ofproto/libofproto.la
+lib_LTLIBRARIES += ofproto/libnoiro_ofproto.la
+ofproto_libnoiro_ofproto_la_SOURCES =
+ofproto_libnoiro_ofproto_la_LIBADD = ofproto/libofproto.la
+
+ofproto_libnoiro_ofproto_la_LDFLAGS = \
         $(OVS_LTINFO) \
         -Wl,--version-script=$(top_builddir)/ofproto/libofproto.sym \
         $(AM_LDFLAGS)
diff --git a/ofproto/libofproto.pc.in b/ofproto/libofproto.pc.in
index 2740712..7a80266 100644
--- a/ofproto/libofproto.pc.in
+++ b/ofproto/libofproto.pc.in
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libofproto
 Description: OpenFlow library of Open vSwitch
 Version: @VERSION@
-Libs: -L${libdir} -lofproto
+Libs: -L${libdir} -lnoiro_ofproto
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/ovn/controller-vtep/automake.mk b/ovn/controller-vtep/automake.mk
index 0c83dc7..2295c84 100644
--- a/ovn/controller-vtep/automake.mk
+++ b/ovn/controller-vtep/automake.mk
@@ -8,7 +8,7 @@ ovn_controller_vtep_ovn_controller_vtep_SOURCES = \
 	ovn/controller-vtep/ovn-controller-vtep.h \
 	ovn/controller-vtep/vtep.c \
 	ovn/controller-vtep/vtep.h
-ovn_controller_vtep_ovn_controller_vtep_LDADD = ovn/lib/libovn.la lib/libopenvswitch.la vtep/libvtep.la
+ovn_controller_vtep_ovn_controller_vtep_LDADD = ovn/lib/libovn.la lib/libnoiro_openvswitch.la vtep/libvtep.la
 man_MANS += ovn/controller-vtep/ovn-controller-vtep.8
 EXTRA_DIST += ovn/controller-vtep/ovn-controller-vtep.8.xml
 CLEANFILES += ovn/controller-vtep/ovn-controller-vtep.8
diff --git a/ovn/controller/automake.mk b/ovn/controller/automake.mk
index 193ea69..7042634 100644
--- a/ovn/controller/automake.mk
+++ b/ovn/controller/automake.mk
@@ -26,7 +26,7 @@ ovn_controller_ovn_controller_SOURCES = \
 	ovn/controller/ovn-controller.h \
 	ovn/controller/physical.c \
 	ovn/controller/physical.h
-ovn_controller_ovn_controller_LDADD = ovn/lib/libovn.la lib/libopenvswitch.la
+ovn_controller_ovn_controller_LDADD = ovn/lib/libovn.la lib/libnoiro_openvswitch.la
 man_MANS += ovn/controller/ovn-controller.8
 EXTRA_DIST += ovn/controller/ovn-controller.8.xml
 CLEANFILES += ovn/controller/ovn-controller.8
diff --git a/ovn/northd/automake.mk b/ovn/northd/automake.mk
index 93aebe8..cccdc48 100644
--- a/ovn/northd/automake.mk
+++ b/ovn/northd/automake.mk
@@ -4,7 +4,7 @@ ovn_northd_ovn_northd_SOURCES = ovn/northd/ovn-northd.c
 ovn_northd_ovn_northd_LDADD = \
 	ovn/lib/libovn.la \
 	ovsdb/libovsdb.la \
-	lib/libopenvswitch.la
+	lib/libnoiro_openvswitch.la
 man_MANS += ovn/northd/ovn-northd.8
 EXTRA_DIST += ovn/northd/ovn-northd.8.xml
 CLEANFILES += ovn/northd/ovn-northd.8
diff --git a/ovn/utilities/automake.mk b/ovn/utilities/automake.mk
index e8c59a2..e0d6950 100644
--- a/ovn/utilities/automake.mk
+++ b/ovn/utilities/automake.mk
@@ -42,16 +42,16 @@ CLEANFILES += \
 # ovn-nbctl
 bin_PROGRAMS += ovn/utilities/ovn-nbctl
 ovn_utilities_ovn_nbctl_SOURCES = ovn/utilities/ovn-nbctl.c
-ovn_utilities_ovn_nbctl_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libopenvswitch.la
+ovn_utilities_ovn_nbctl_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 
 # ovn-sbctl
 bin_PROGRAMS += ovn/utilities/ovn-sbctl
 ovn_utilities_ovn_sbctl_SOURCES = ovn/utilities/ovn-sbctl.c
-ovn_utilities_ovn_sbctl_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libopenvswitch.la
+ovn_utilities_ovn_sbctl_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 
 # ovn-trace
 bin_PROGRAMS += ovn/utilities/ovn-trace
 ovn_utilities_ovn_trace_SOURCES = ovn/utilities/ovn-trace.c
-ovn_utilities_ovn_trace_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libopenvswitch.la
+ovn_utilities_ovn_trace_LDADD = ovn/lib/libovn.la ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 
 include ovn/utilities/bugtool/automake.mk
diff --git a/ovsdb/automake.mk b/ovsdb/automake.mk
index b895f42..594b422 100644
--- a/ovsdb/automake.mk
+++ b/ovsdb/automake.mk
@@ -59,7 +59,7 @@ MAN_FRAGMENTS += ovsdb/ovsdb-schemas.man
 # ovsdb-tool
 bin_PROGRAMS += ovsdb/ovsdb-tool
 ovsdb_ovsdb_tool_SOURCES = ovsdb/ovsdb-tool.c
-ovsdb_ovsdb_tool_LDADD = ovsdb/libovsdb.la lib/libopenvswitch.la
+ovsdb_ovsdb_tool_LDADD = ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 # ovsdb-tool.1
 man_MANS += ovsdb/ovsdb-tool.1
 CLEANFILES += ovsdb/ovsdb-tool.1
@@ -68,7 +68,7 @@ MAN_ROOTS += ovsdb/ovsdb-tool.1.in
 # ovsdb-client
 bin_PROGRAMS += ovsdb/ovsdb-client
 ovsdb_ovsdb_client_SOURCES = ovsdb/ovsdb-client.c
-ovsdb_ovsdb_client_LDADD = ovsdb/libovsdb.la lib/libopenvswitch.la
+ovsdb_ovsdb_client_LDADD = ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 # ovsdb-client.1
 man_MANS += ovsdb/ovsdb-client.1
 CLEANFILES += ovsdb/ovsdb-client.1
@@ -77,7 +77,7 @@ MAN_ROOTS += ovsdb/ovsdb-client.1.in
 # ovsdb-server
 sbin_PROGRAMS += ovsdb/ovsdb-server
 ovsdb_ovsdb_server_SOURCES = ovsdb/ovsdb-server.c
-ovsdb_ovsdb_server_LDADD = ovsdb/libovsdb.la lib/libopenvswitch.la
+ovsdb_ovsdb_server_LDADD = ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 # ovsdb-server.1
 man_MANS += ovsdb/ovsdb-server.1
 CLEANFILES += ovsdb/ovsdb-server.1
diff --git a/tests/automake.mk b/tests/automake.mk
index d6ab517..15e660c 100644
--- a/tests/automake.mk
+++ b/tests/automake.mk
@@ -396,7 +396,7 @@ $(srcdir)/package.m4: $(top_srcdir)/configure.ac
 noinst_PROGRAMS += tests/test-ovsdb
 tests_test_ovsdb_SOURCES = tests/test-ovsdb.c
 nodist_tests_test_ovsdb_SOURCES = tests/idltest.c tests/idltest.h
-tests_test_ovsdb_LDADD = ovsdb/libovsdb.la lib/libopenvswitch.la
+tests_test_ovsdb_LDADD = ovsdb/libovsdb.la lib/libnoiro_openvswitch.la
 
 noinst_PROGRAMS += tests/test-lib
 tests_test_lib_SOURCES = \
@@ -474,7 +474,7 @@ tests_ovstest_SOURCES += \
 	tests/test-netlink-conntrack.c
 endif
 
-tests_ovstest_LDADD = lib/libopenvswitch.la ovn/lib/libovn.la
+tests_ovstest_LDADD = lib/libnoiro_openvswitch.la ovn/lib/libovn.la
 
 noinst_PROGRAMS += tests/test-stream
 tests_test_stream_SOURCES = tests/test-stream.c
diff --git a/vtep/automake.mk b/vtep/automake.mk
index 0f313dc..e350fbf 100644
--- a/vtep/automake.mk
+++ b/vtep/automake.mk
@@ -34,7 +34,7 @@ man_MANS += \
    vtep/vtep-ctl.8
 
 vtep_vtep_ctl_SOURCES = vtep/vtep-ctl.c
-vtep_vtep_ctl_LDADD = vtep/libvtep.la lib/libopenvswitch.la
+vtep_vtep_ctl_LDADD = vtep/libvtep.la lib/libnoiro_openvswitch.la
 
 # ovs-vtep
 scripts_SCRIPTS += \
